Momento/MomentoSigner.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.IdentityModel.Tokens.Jwt;
+using Microsoft.IdentityModel.Tokens;
+using MomentoSdk.Exceptions;
+
+namespace MomentoSdk
+{
+    public class MomentoSigner
+    {
+
+        /// <summary>
+        /// Create a pre-signed HTTPS URL to retrieve an object.
+        /// </summary>
+        /// <param name="hostname">Hostname of the SimpleCacheService. Use the value returned from CreateSigningKey's response.</param>
+        /// <param name="jwkJsonString">JWK in JSON string. Use the value returned from CreateSigningKey's response.</param>
+        /// <param name="cacheName">The name of the Cache.</param>
+        /// <param name="cacheKey">The key of the Object.</param>
+        /// <param name="expiryEpochSeconds">The timestamp that the pre-signed URL is valid until.</param>
+        /// <returns></returns>
+        public string CreatePresignedUrlForGet(string hostname, string jwkJsonString, string cacheName, string cacheKey, uint expiryEpochSeconds)
```

Comment:
Looking at the accessibility modifiers for this method. It seems like customers will be calling this method directly. How are they going to provide the hostname? The hostname is tied into our Momento Auth token and customers have no knowledge yet about how to parse that out.

Reply:
I am also quite curious as to why this isn't a method in `SimpleCacheClient`?

Reply:
re: why this isn't a method in SimpleCacheClient

https://gomomento.slack.com/archives/C01AC7JT370/p1649105256165299?thread_ts=1649103168.349359&cid=C01AC7JT370

Reply:
> The hostname is tied into our Momento Auth token

This is an intentional design choice for the pre-signed url implementation. We are going to return the hostname as a part of the CreateSigningKeyResponse: https://github.com/momentohq/client-sdk-rust/blob/main/src/response/create_signing_key_response.rs#L8

Reply:
So after creating a signing key, customers will need to tie (via code or configuration) signing key and the hostname in someway. This allows us to decouple the pre-signed url endpoints from the cache service. Just that we are putting onus on customers now. 

I see pros and cons to both approaches. We can start with this and learn from feedback 👍